tools/checked-c-convert/CheckedCConvert.cpp

```diff
@@ -732,64 +732,84 @@ bool CastPlacementVisitor::VisitFunctionDecl(FunctionDecl *FD) {
 
   if (cDecl->numParams() == cDefn->numParams()) { 
     bool didAny = false;
-		std::string s = "";
-		std::vector<std::string> parmStrs;
-		// Compare parameters. 
-		if (cDecl->numParams() == cDefn->numParams())
-			for (unsigned i = 0; i < cDecl->numParams(); ++i) {
-				auto Decl = getHighest(cDecl->getParamVar(i), Info);
-				auto Defn = getHighest(cDefn->getParamVar(i), Info);
-				assert(Decl);
-				assert(Defn);
-
-				// If this holds, then we want to insert a bounds safe interface. 
-				if (Defn->isLt(*Decl, Info)) {
-					std::string scratch = "";
-					raw_string_ostream declText(scratch);
-					Definition->getParamDecl(i)->print(declText);
-					std::string ctype = Defn->mkString(Info.getConstraints().getVariables(), false);
-					std::string bi = declText.str() + " : itype("+ctype+") ";
-					parmStrs.push_back(bi);
-					didAny = true;
-				} else {
-					// Do what we used to do. 
-					std::string v = Decl->mkString(Info.getConstraints().getVariables());
-					parmStrs.push_back(v);
-				}
-			}
-
-		// Compare returns. 
-		auto Decl = getHighest(cDecl->getReturnVars(), Info);
-		auto Defn = getHighest(cDefn->getReturnVars(), Info);
-
-		// Insert a bounds safe interface at the return address. 
-		std::string returnVar = "";
-		std::string endStuff = "";
-		if (Defn->isLt(*Decl, Info)) {
-			std::string ctype = Defn->mkString(Info.getConstraints().getVariables());
-			returnVar = Defn->getTy();
-			endStuff = " : itype("+ctype+") ";
-			didAny = true;
-		} else {
-			// Do what we used to do at the return address. 
-			returnVar = Decl->mkString(Info.getConstraints().getVariables()) + " ";
-		}
-
-		s = returnVar + cDecl->getName() + "(";
-		if (parmStrs.size() > 0) {
-			std::ostringstream ss;
-
-			std::copy(parmStrs.begin(), parmStrs.end() - 1,
-					 std::ostream_iterator<std::string>(ss, ", "));
-			ss << parmStrs.back();
-
-			s = s + ss.str() + ")";
-		} else {
-			s = s + "void)";
-		}
-
-		if (endStuff.size() > 0)
-			s = s + endStuff;
+    std::string s = "";
+    std::vector<std::string> parmStrs;
+    // Compare parameters. 
+    if (cDecl->numParams() == cDefn->numParams())
+      for (unsigned i = 0; i < cDecl->numParams(); ++i) {
+        auto Decl = getHighest(cDecl->getParamVar(i), Info);
+        auto Defn = getHighest(cDefn->getParamVar(i), Info);
+        assert(Decl);
+        assert(Defn);
+
+        didAny = true;
```

Comment:
I don't understand why you moved the assignment to didAny to here.   This seems to always set it.

Reply:
Roughly, there's now only one case where we won't re-write a declaration in this function - which is when there is a parameter mis-match indicating we don't exactly know what to do. I should remove `didAny` and replace it with something more sensible.